src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -1193,15 +1193,24 @@ private BoundExpression MakeNullableHasValue(SyntaxNode syntax, BoundExpression
             BoundExpression consequence;
             if (operatorKind == BinaryOperatorKind.Equal || operatorKind == BinaryOperatorKind.NotEqual)
             {
-                // tempx.HasValue ? tempX.GetValueOrDefault() == tempY.GetValueOrDefault() : true
-                consequence = RewriteConditionalOperator(
-                    syntax: syntax,
-                    rewrittenCondition: callX_HasValue,
-                    rewrittenConsequence: unliftedOp,
-                    rewrittenAlternative: MakeLiteral(syntax, ConstantValue.Create(operatorKind == BinaryOperatorKind.Equal), boolType),
-                    constantValueOpt: null,
-                    rewrittenType: boolType,
-                    isRef: false);
+                // If one operand is known to be non-null, we would get `x.HasValue ? (x.HasValue ? x.GVOD() OP y.GVOD() : true) : false`.
```

Comment:
Is there a `==` missing from this comment?

Reply:
Technically it can be `==` or `!=` so the comment has `OP` instead.

Reply:
Ah, ok. I think that broke my brain a bit, but makes sense reading it again.